### PR TITLE
Mock ajax calls for testing

### PR DESCRIPTION
**Pull request by donoftime**
I saw your suggestions across all the pull requests (and I'm working on them!) but I figured I would start on this one so I don't have to keep turning on my web server. 

I added sample testdata so that:

1. New contributors would hit the ground running with data without having to collect their own
1. It would be obvious how to add *new* testdata whenever someone had something worth testing. For example, we're going to need a year's worth of dailies and todos when we get to those pages.

I just brought in a random library that I came across to make this happen. Normally I would have used something like Jasmine or Sinon.js for this, but I didn't want to bring in all that just to mock ajax calls - especially since we're doing them all with jquery anyway. I figured we could just use the jquery mock and call it good :sweat_smile:

### CODE REVIEW

**Comment by donoftime** [comment]
Oh yeah, and I played around with the IDs and hashes I came across in the sample data so they would look right - but not actually be right.

File: habitrpg_user_data_display.html [hunk 1]
```diff
@@ -56,6 +56,15 @@
     <script src="js/DataTables/extras/ColumnFilterWidgets/media/js/ColumnFilterWidgets.js"></script>
     <script src="js/emoji-images/emoji-images.js"></script>
 
+    <!-- TST
```
Comment (Alys) [inline habitrpg_user_data_display.html]: Could you make this "TST - FOR TESTING:"?  "TST" by itself is a code I use for temporary test lines that are usually removed before going live; "TST - FOR TESTING" is for permanent test lines (including "TST" means that I can find both kinds of test lines with a single search).

File: habitrpg_user_data_display.html [hunk 2]
```diff
@@ -56,6 +56,15 @@
...
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-mockjax/1.5.3/jquery.mockjax.min.js" type="text/javascript"></script>
```
Comment (Alys) [inline habitrpg_user_data_display.html]: Can you add a comment with the URL of the site that describes this library?

File: habitrpg_user_data_display.html [hunk 3]
```diff
@@ -82,16 +91,29 @@
...
 $('body').css({'background-color':'#C4B4A4'});
-serverName        = 'localhost';
-serverUrl         = 'http://localhost';
-serverPathTavern  = '/cgi-bin/hrpg_tavern.pl';
-serverPathContent = '/cgi-bin/hrpg_content.pl';
-serverPathUser    = '/cgi-bin/hrpg_user.pl';
 apiToken          = 'null';
 userId            = 'test';
-userId            = 'work';
-userId            = 'home';
```
Comment (Alys) [inline habitrpg_user_data_display.html]: I can definitely see that the mockajax calls make it much easier for devs to use custom test data, and I think that's a great idea. However I still want to use my localhost scripts with my home and work data for my own testing (I have interrelated systems set up for this and other things that are working well for me). Can you please put the new code inside a separate "TST - FOR TESTING" comment and undelete my test code? I recommend including the background colour change in both, but it should at least be in mine.

**Comment by donoftime** [comment]
Sorry I deleted your webserver configurations. I was trying to pick up behind myself and got a little carried away :laughing:

**Comment by Alys** [comment]
No worries. :)  If I didn't already have a system of my own, I would have just gone with your new way.

**Comment by Alys** [comment]
Thank you for this! If you give me your UUID, you'll get Tier 1 for this.

**Comment by donoftime** [comment]
Cool! I'm glad it's helpful :)
9f189c6a-2db9-4e89-b710-ee8489415cc5

**Pull request merged**